arrow/array/dictionary.go

```diff
@@ -1402,143 +1107,178 @@ func (b *BinaryDictionaryBuilder) ValueStr(i int) string {
-		if err = b.insertDictValue(data[:arrow.Decimal128SizeBytes]); err != nil {
-			break
-		}
-		data = data[arrow.Decimal128SizeBytes:]
+func (b *fixedSizeDictionaryBuilder[T]) Append(v T) error {
+	if v, ok := any(v).([]byte); ok {
+		return b.appendBytes(v[:b.byteWidth])
 	}
-	return
-}
-
-type Decimal256DictionaryBuilder struct {
-	dictionaryBuilder
-}
 
-func (b *Decimal256DictionaryBuilder) Append(v decimal256.Num) error {
-	return b.appendValue((*(*[arrow.Decimal256SizeBytes]byte)(unsafe.Pointer(&v)))[:])
+	sliceHdr := struct {
+		Addr *T
+		Len  int
+		Cap  int
+	}{&v, b.byteWidth, b.byteWidth}
+	slice := *(*[]byte)(unsafe.Pointer(&sliceHdr))
+	return b.appendValue(slice)
 }
 
-func (b *Decimal256DictionaryBuilder) InsertDictValues(arr *Decimal256) (err error) {
-	data := arrow.Decimal256Traits.CastToBytes(arr.values)
+func (b *fixedSizeDictionaryBuilder[T]) InsertDictValues(arr arrValues[T]) (err error) {
+	data := arrow.GetBytes(arr.Values())
 	for len(data) > 0 {
-		if err = b.insertDictValue(data[:arrow.Decimal256SizeBytes]); err != nil {
+		if err = b.insertDictBytes(data[:b.byteWidth]); err != nil {
 			break
 		}
-		data = data[arrow.Decimal256SizeBytes:]
+		data = data[b.byteWidth:]
 	}
 	return
 }
 
-type MonthDayNanoDictionaryBuilder struct {
+type FixedSizeBinaryDictionaryBuilder struct {
 	dictionaryBuilder
+	byteWidth int
 }
 
-func (b *MonthDayNanoDictionaryBuilder) Append(v arrow.MonthDayNanoInterval) error {
-	return b.appendValue((*(*[arrow.MonthDayNanoIntervalSizeBytes]byte)(unsafe.Pointer(&v)))[:])
+func (b *FixedSizeBinaryDictionaryBuilder) Append(v []byte) error {
+	return b.appendValue(v[:b.byteWidth])
 }
 
-func (b *MonthDayNanoDictionaryBuilder) InsertDictValues(arr *MonthDayNanoInterval) (err error) {
-	data := arrow.MonthDayNanoIntervalTraits.CastToBytes(arr.values)
+func (b *FixedSizeBinaryDictionaryBuilder) InsertDictValues(arr *FixedSizeBinary) (err error) {
+	var (
+		beg = arr.array.data.offset * b.byteWidth
+		end = (arr.array.data.offset + arr.data.length) * b.byteWidth
+	)
+	data := arr.valueBytes[beg:end]
 	for len(data) > 0 {
-		if err = b.insertDictValue(data[:arrow.MonthDayNanoIntervalSizeBytes]); err != nil {
+		if err = b.insertDictValue(data[:b.byteWidth]); err != nil {
 			break
 		}
-		data = data[arrow.MonthDayNanoIntervalSizeBytes:]
+		data = data[b.byteWidth:]
 	}
 	return
 }
 
-type DayTimeDictionaryBuilder struct {
-	dictionaryBuilder
-}
-
-func (b *DayTimeDictionaryBuilder) Append(v arrow.DayTimeInterval) error {
-	return b.appendValue((*(*[arrow.DayTimeIntervalSizeBytes]byte)(unsafe.Pointer(&v)))[:])
-}
-
-func (b *DayTimeDictionaryBuilder) InsertDictValues(arr *DayTimeInterval) (err error) {
-	data := arrow.DayTimeIntervalTraits.CastToBytes(arr.values)
-	for len(data) > 0 {
-		if err = b.insertDictValue(data[:arrow.DayTimeIntervalSizeBytes]); err != nil {
-			break
-		}
-		data = data[arrow.DayTimeIntervalSizeBytes:]
-	}
-	return
-}
+// type Decimal32DictionaryBuilder struct {
```

Comment:
Did you mean to leave this code commented out?

Reply:
Nope, oops. Thanks!